src/metabase/driver/common/parameters/dates.clj

```diff
@@ -172,34 +172,35 @@
    {:parser (regex->parser (re-pattern (str #"past([0-9]+)" temporal-units-regex #"s" relative-suffix-regex))
                            [:int-value :unit :relative-suffix :int-value-1 :unit-1])
     :range  (fn [{:keys [unit int-value unit-range to-period relative-suffix unit-1 int-value-1]} dt]
-              (let [dt-off (cond-> dt
-                             unit-1 (t/minus ((get-in operations-by-date-unit [unit-1 :to-period]) int-value-1)))
-                    dt-res (maybe-reduce-resolution unit dt-off)]
-                (unit-range (t/minus dt-res (to-period int-value))
-                            (t/minus dt-res (to-period (if (seq relative-suffix) 0 1))))))
+              (let [dt-offset (cond-> dt
+                                unit-1 (t/minus ((get-in operations-by-date-unit [unit-1 :to-period]) int-value-1)))
+                    dt-resolution (maybe-reduce-resolution unit dt-offset)]
+                (unit-range (t/minus dt-resolution (to-period int-value))
+                            (t/minus dt-resolution (to-period (if (= "~" relative-suffix) 0 1))))))
+
     :filter (fn [{:keys [unit int-value relative-suffix unit-1 int-value-1]} field-clause]
               (if unit-1
                 [:between
                  [:+ field-clause [:interval int-value-1 (keyword unit-1)]]
                  [:relative-datetime (- int-value) (keyword unit)]
                  [:relative-datetime 0 (keyword unit)]]
-                [:time-interval field-clause (- int-value) (keyword unit) {:include-current (boolean (seq relative-suffix))}]))}
+                [:time-interval field-clause (- int-value) (keyword unit) {:include-current (= "~" relative-suffix)}]))}
 
    {:parser (regex->parser (re-pattern (str #"next([0-9]+)" temporal-units-regex #"s" relative-suffix-regex))
                            [:int-value :unit :relative-suffix :int-value-1 :unit-1])
     :range  (fn [{:keys [unit int-value unit-range to-period relative-suffix unit-1 int-value-1]} dt]
-              (let [dt-off (cond-> dt
-                             unit-1 (t/plus ((get-in operations-by-date-unit [unit-1 :to-period]) int-value-1)))
-                    dt-res (maybe-reduce-resolution unit dt-off)]
-                (unit-range (t/plus dt-res (to-period (if (seq relative-suffix) 0 1)))
-                            (t/plus dt-res (to-period int-value)))))
+              (let [dt-offset (cond-> dt
+                                unit-1 (t/plus ((get-in operations-by-date-unit [unit-1 :to-period]) int-value-1)))
+                    dt-resolution (maybe-reduce-resolution unit dt-offset)]
+                (unit-range (t/plus dt-resolution (to-period (if (= "~" relative-suffix) 0 1)))
```

Comment:
I've seen the `(= relative-suffix "~")` check in a few places... is it worth abstracting out into a separate function? Not so much to save code duplication, but to make the code easier to understand. Something like `(include-current-suffix? relative-suffix)` would be easier for people to understand than `(= "~" ...)`. Not everybody might know the special purpose of `~` which could make this confusing